substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/NativeImageGenerator.java

```diff
@@ -565,31 +565,31 @@ private void doRun(Map<Method, CEntryPointData> entryPoints,
                     throw UserError.abort("Warning: no entry points found, i.e., no method annotated with @" + CEntryPoint.class.getSimpleName());
                 }
 
+                bigbang.getUnsupportedFeatures().report(bigbang);
+
+                recordMethodsWithStackValues();
+                recordRestrictHeapAccessCallees(aUniverse.getMethods());
+
+                /*
+                 * After this point, all TypeFlow (and therefore also TypeState) objects are
+                 * unreachable and can be garbage collected. This is important to keep the overall
+                 * memory footprint low. However, this also means we no longer have complete call
+                 * chain information. Only the summarized information stored in the
+                 * StaticAnalysisResult objects is available after this point.
+                 */
+                bigbang.cleanupAfterAnalysis();
+
                 ImageHeapLayouter heapLayouter = ImageSingletons.lookup(ImageHeapLayouter.class);
                 heap = new NativeImageHeap(aUniverse, hUniverse, hMetaAccess, heapLayouter);
 
                 BeforeCompilationAccessImpl config = new BeforeCompilationAccessImpl(featureHandler, loader, aUniverse, hUniverse, hMetaAccess, heap, debug, runtime);
                 featureHandler.forEachFeature(feature -> feature.beforeCompilation(config));
 
                 runtime.updateLazyState(hMetaAccess);
-
-                bigbang.getUnsupportedFeatures().report(bigbang);
```

Comment:
This line should remain the last one before the `catch`, otherwise we could miss unsupported features that get accumulated in the `UnsupportedFeatures` object.

Reply:
The `ShortestInvokeChainPrinter` used by the unsupported features reporting needs `AnalysisMethod.typeFlow` which is cleaned up by  `bigbang.cleanupAfterAnalysis()`. I think we should actually move everything after ` bigbang.cleanupAfterAnalysis();` outside of the catch, `beforeCompilation` should not accumulate unspported features.

Reply:
Yes, sounds good